akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -922,7 +934,13 @@ private[akka] final class FunctionRef(
     }
   }
 
-  override def stop(): Unit = sendTerminated()
+  override def stop(): Unit = {
+    sendTerminated()
+    // The messageHandler function may close over a large object graph (such as an Akka Stream)
+    // so we replace the messageHandler function to make that available for garbage collection.
+    // Doesn't matter if the change isn't visible immediately, volatile not needed.
+    messageHandler = FunctionRef.deadLetterMessageHandler(system)
```

Comment:
can deadLetterMessageHandler be cached?

Reply:
That would be by placing it in an Akka extension, since it's tied to one ActorSystem. Saving one function allocation at the price of a ConcurrentHashMap lookup. Not worth it.